Missing options for `zig c++ -undefined` parameter (warning, error, suppress)

### Zig Version

0.11.0-dev.927+71668fc4e

### Steps to Reproduce and Observed Behavior

```shell
echo 'int main() { return 0; }' > /tmp/test.cc
zig-macos-aarch64-0.11.0-dev.927+71668fc4e/zig c++ -o /tmp/test /tmp/test.cc -undefined error
```

```
error: unsupported -undefined option 'error'
```

### Expected Behavior

I ran into this issue when attempting to use Bazel with bazel-zig-cc and rules_foreign_cc. On macos, the latter passes -undefined error when building GNU make [1].

zig 0.10 accepts this option with a warning, so `rules_foreign_cc` has worked.
```
zig: warning: argument unused during compilation: '-undefined error' [-Wunused-command-line-argument]
```

It looks like clang passes this parameter through to the linker, and on macos the linker accepts `warning`, `error`, and `suppress` in addition to `dynamic_lookup` [2].

I believe zig 0.11's behavior changed in https://github.com/ziglang/zig/pull/13991. Will support for these other options be added? If not, I'll try to patch Bazel's `rules_foreign_cc`.

1:  https://github.com/bazelbuild/rules_foreign_cc/blob/baeee718db3ce2c9c1861e74a5f56c3bab71e9e3/foreign_cc/built_tools/make_build.bzl#L75
2: https://github.com/llvm-mirror/lld/blob/64b024a57c56c3528d6be3d14be5e3da42614a6f/lib/Driver/DarwinLdDriver.cpp#L996

Thanks @kubkon.

I'm not an expert in these options; I'm just trying to connect the dots between `zig c++` and Bazel's `rules_foreign_cc`.

That said, after some digging, I think adding support for just `error` in addition to `dynamic_lookup` would be sufficient. `error` is just the default behavior as far as I can tell, so handling it would allow a progressively-built list of cflags to reset back to the default.

For example:

foo.cc:
```c++
extern int foo() {
    return 3;
}
```

bar.cc:
```c++
extern int foo();

int bar() { return foo(); }
```

So with clang++, a later `-Wl,-undefined,error` can override an earlier `-Wl,-undefined,dynamic_lookup`:
```
# success, because foo.cc is included
➜  /tmp clang++ -shared -Wl,-undefined,dynamic_lookup -Wl,-undefined,error foo.cc bar.cc -o bar

# error
➜  /tmp clang++ -shared -Wl,-undefined,dynamic_lookup -Wl,-undefined,error bar.cc -o bar 
Undefined symbols for architecture arm64:
  "foo()", referenced from:
      bar() in bar-fea150.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

But with `zig c++`, both just fail because `error` is unsupported.
```
➜  /tmp zig c++ -shared -Wl,-undefined,dynamic_lookup -Wl,-undefined,error foo.cc bar.cc -o bar
error: unsupported -undefined option 'error'
➜  /tmp zig c++ -shared -Wl,-undefined,dynamic_lookup -Wl,-undefined,error bar.cc -o bar 
error: unsupported -undefined option 'error'
```

--- a/src/main.zig
+++ b/src/main.zig
@@ -2073,6 +2073,8 @@ fn buildOutputType(
                     }
                     if (mem.eql(u8, "dynamic_lookup", linker_args.items[i])) {
                         linker_allow_shlib_undefined = true;
+                    } else if (mem.eql(u8, "error", linker_args.items[i])) {
+                        linker_allow_shlib_undefined = false;
                     } else {
                         fatal("unsupported -undefined option '{s}'", .{linker_args.items[i]});
                     }

--- a/src/main.zig
+++ b/src/main.zig
@@ -1703,6 +1703,8 @@ fn buildOutputType(
                     .undefined => {
                         if (mem.eql(u8, "dynamic_lookup", it.only_arg)) {
                             linker_allow_shlib_undefined = true;
+                        } else if (mem.eql(u8, "error", it.only_arg)) {
+                            linker_allow_shlib_undefined = false;
                         } else {
                             fatal("unsupported -undefined option '{s}'", .{it.only_arg});
                         }

